MultiplayerExtensions/UI/MpexGameplaySetup.cs

```diff
@@ -0,0 +1,87 @@
+﻿using BeatSaberMarkupLanguage;
+using BeatSaberMarkupLanguage.Attributes;
+using IPA.Utilities;
+using SiraUtil.Logging;
+using System;
+using System.Reflection;
+using UnityEngine;
+using Zenject;
+
+namespace MultiplayerExtensions.UI
+{
+    public class MpexGameplaySetup : IInitializable, IDisposable
```

Comment:
also inherit NotifiableBase from BSML then in your set functions call NotifyPropertyChanged like this https://github.com/rithik-b/PlaylistManager/blob/ae661cffdb65e39dc928a15f4fdfec9740f0221f/PlaylistManager/UI/ViewControllers/ImageSelectionModalController.cs#L227